### PR TITLE
[MM-52823] Migration of "components/admin_console/custom_url_schemes_setting.jsx" and tests to typescript

### PR DESCRIPTION
**Pull request by ujwalkumar1995**
#### Summary
This PR if for migration of "components/admin_console/custom_url_schemes_setting.jsx" and corresponding tests to typescript. Have updated the source file, tests and snapshot tests. Also updated the path where it is used.

#### Ticket Link
Github Issue: #23437 
JIRA Ticket: https://mattermost.atlassian.net/browse/MM-52823

#### Screenshots
N/A

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ import ExternalLink from 'components/external_link';
 
 import OpenIdConvert from './openid_convert';
 import Audits from './audits';
-import CustomURLSchemesSetting from './custom_url_schemes_setting.jsx';
+import CustomURLSchemesSetting from './custom_url_schemes_setting.tsx';
```
Comment (sinansonmez) [inline webapp/channels/src/components/admin_console/admin_definition.jsx:39]: I believe you dont need to explicitly mention `tsx`

**Review by sinansonmez — APPROVED** [review]
Just one minor comment. Otherwise LGTM

File: webapp/channels/src/components/admin_console/admin_definition.jsx (hunk 1 above)
Comment (ujwalkumar1995) [reply webapp/channels/src/components/admin_console/admin_definition.jsx]: Thank you for the review. Updated the import.

File: webapp/channels/src/components/admin_console/custom_url_schemes_setting.tsx [hunk 2]
```diff
@@ -11,7 +11,20 @@ import LocalizedInput from 'components/localized_input/localized_input';
 
 import Setting from './setting';
 
-export default class CustomURLSchemesSetting extends React.PureComponent {
+type CustomURLSchemesSettingProps = {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/custom_url_schemes_setting.tsx:14]: nit : rename this to just `Props`

File: webapp/channels/src/components/admin_console/custom_url_schemes_setting.tsx [hunk 3]
```diff
@@ -11,7 +11,20 @@ import LocalizedInput from 'components/localized_input/localized_input';
...
+    id: string;
+    value: string[];
+    onChange: (id: string, valueAsArray: string[]) => void;
+    disabled: boolean;
+    setByEnv: boolean;
+}
+
+type CustomURLSchemesSettingState = {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/custom_url_schemes_setting.tsx:22]: nit : rename this to just State

File: webapp/channels/src/components/admin_console/custom_url_schemes_setting.tsx [hunk 4]
```diff
@@ -11,7 +11,20 @@ import LocalizedInput from 'components/localized_input/localized_input';
...
+    value: string;
+}
+
+export default class CustomURLSchemesSetting extends
+    Component<CustomURLSchemesSettingProps, CustomURLSchemesSettingState> {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/custom_url_schemes_setting.tsx:27]: this was extended from PureComponents

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Comment by M-ZubairAhmed** [comment]
@yasserfaraazkhan I think we need to open a branch from this on upstream and run the e2e test. Because all opened ones are failing consistently.

**Comment by M-ZubairAhmed** [comment]
> e2e passed now @M-ZubairAhmed can you help in cross checking the failed one related to plugin ?

I think these are unrelated. merging thank you

**Pull request merged**